### PR TITLE
[frontend] added pop ups and connected note and discussion to backend

### PR DESCRIPTION
**Pull request by mertcan-ozkan**
- connected delete add update note
- connected add discussion
- created pop ups 
- also added a button called add notes to resource file.

### CODE REVIEW

File: app/client/pages/my/spaces/[space_id]/discussion/[discussion_id].js [hunk 1]
```diff
@@ -4,79 +4,57 @@ import "@uiw/react-md-editor/markdown-editor.css";
...
+    useEffect(() => {
+        discussion = router.query;
+        fetchDiscussion();
+    }, [discussion.discussion_id]);
+
+    useEffect(() => { }, [previousComments])
 
     return (
         <section className={styles.container}>
             <h2>Acoustic Guitar Ed for Beginners</h2>
```
Comment (berkeozdemir2016400246) [inline app/client/pages/my/spaces/[space_id]/discussion/[discussion_id].js:46]: Can change this to

File: app/client/pages/my/spaces/[space_id]/notes.js [hunk 2]
```diff
@@ -1,46 +1,63 @@
...
+    useEffect(() => {
+       
+        fetchNotes();
+    }, [openEditNote]);
+  
     return (
         <section className={styles.container}>
+
+
             <h2>Acoustic Guitar Ed for Beginners</h2>
```
Comment (berkeozdemir2016400246) [inline app/client/pages/my/spaces/[space_id]/notes.js:46]: This should be changed to space title

File: app/client/pages/my/spaces/[space_id]/resource/[resource].js [hunk 3]
```diff
@@ -72,11 +77,14 @@ export default function resource() {
                         <span>{new Date(data?.resource?.createdAt).toLocaleDateString()}</span>
 
                     </div>
+                    <div className={styles.resourceDetailHeader}>
```
Comment (berkeozdemir2016400246) [inline app/client/pages/my/spaces/[space_id]/resource/[resource].js:80]: This needs a left margin

File: app/client/components/Note/note.jsx [hunk 4]
```diff
@@ -8,43 +8,69 @@ import DeleteIcon from '@mui/icons-material/Delete';
...
-                        <Button><ExitToAppIcon /></Button>
-                        <Button><EditIcon /></Button>
-                        <Button><ShareIcon /></Button>
-                        <Button onClick={deleteNote}><DeleteIcon /></Button>
-                    </ButtonGroup>
-                </CardActions>
-            </Collapse>
-        </Card>
+   
+            <Card onMouseEnter={() => setActions(true)} onMouseLeave={() => setActions(false)}>
```
Comment (berkeozdemir2016400246) [inline app/client/components/Note/note.jsx:40]: I think we should add name of the resource notes belong to somewhere here

**Review by berkeozdemir2016400246 — CHANGES_REQUESTED** [review]
Good work, I have commented some minor issues with the UI

File: app/client/components/Note/note.jsx (hunk 4 above)
Comment (mertcan-ozkan) [reply app/client/components/Note/note.jsx:40]: You are right. But right now our notes have no title. I think we should leave it as it for the moment. Also in note creation pop up title is just set to "" for the moment.

File: app/client/pages/my/spaces/[space_id]/resource/[resource].js (hunk 3 above)
Comment (mertcan-ozkan) [reply app/client/pages/my/spaces/[space_id]/resource/[resource].js:80]: I think nurlan is implementing this page. I just added a button for note creation.There might be merging issues.

File: app/client/pages/my/spaces/[space_id]/discussion/[discussion_id].js (hunk 1 above)
Comment (mertcan-ozkan) [reply app/client/pages/my/spaces/[space_id]/discussion/[discussion_id].js:61]: fixed.

**Review by berkeozdemir2016400246 — APPROVED** [review]

**Pull request merged**